config/v1/types_infrastructure.go

```diff
@@ -144,6 +148,21 @@ type GCPPlatformStatus struct {
 	Region string `json:"region"`
 }
 
+// BareMetalPlatformStatus holds the current status of the BareMetal infrastructure provider.
+type BareMetalPlatformStatus struct {
+	// apiVIP holds the Virtual IP (VIP) that can be used to contact the OpenShift API.
+	// It is for a self-hosted load balancer in front of the API servers.
+	APIVIP string `json:"apiVIP"`
```

Comment:
What is the advantage of this being described as a "Virtual IP" from a consumer API perspective rather than just an IP? Is it always an IP, or can it be a hostname?

Reply:
AFAIK, it's just an IP. We can remove the "Virtual" part of it, but it's being set by the installer, and consumed by MCO (eventually) to configure the static pods being set up for BareMetal.

Reply:
It's clear we lack good documentation that explains all of this.  @bcrochet can you please ensure that documentation of the DNS and load balancing work is on your team's backlog?  cc/ @celebdor @cybertron @yboaron 

To the question ... this is a configuration item needed for IPI bare metal clusters.  This IP is used to font a self hosted load balancer that's in front of the API servers.  It will never be a hostname.  It's an IP address reserved on the network for our use.